lib/ansible/modules/cloud/google/gcp_compute_firewall.py

```diff
@@ -445,7 +445,14 @@ def main():
             source_tags=dict(type='list', elements='str'),
             target_service_accounts=dict(type='list', elements='str'),
             target_tags=dict(type='list', elements='str')
-        )
+        ),
+        mutually_exclusive=[['allowed', 'denied'],
+                            ['destination_ranges', 'source_ranges', 'source_tags'],
+                            ['destination_ranges', 'source_ranges'],
+                            ['source_service_accounts', 'source_tags', 'target_tags'],
+                            ['destination_ranges', 'source_service_accounts', 'source_tags', 'target_service_accounts'],
```

Comment:
wouldn't just this line (+ allowed/denied) suffice? or is this this way because it's easier to generate?

Reply:
Yes, this is very true. Looking into a fix now

(Also, this feels like an amazing interview question...)

Reply:
I started doing some work into this.

Getting rid of the duplicates is quite the pain. I'm a little inclined to keep it as-is unless Ansible tells me otherwise.

Thoughts?

Reply:
Sure, that's fine! I'd just make sure that it's documented somewhere that we do it this way because it's easier to code vs because they have to be separated in that way.

Reply:
I documented it somewhere in the MM code. Think it's worth putting directly in the Ansible module as well?

Reply:
MM code is fine, thanks!